src/gui/wizard/welcomepage.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2021 by Felix Weilbach <felix.weilbach@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include <QWizardPage>
+
+#include "ui_welcomepage.h"
+#include "wizard/owncloudwizardcommon.h"
+
+namespace OCC {
+
+class OwncloudWizard;
+
+class WelcomePage : public QWizardPage
+{
+    Q_OBJECT
+
+public:
+    WelcomePage(OwncloudWizard *ocWizard);
+    int nextId() const override;
+    void initializePage() override;
+    void setLoginButtonDefault(bool value = true);
+
+    Ui_WelcomePage _ui;
```

Comment:
Thank you for catching that. This is a left over from a long debugging session that I overlooked.
Are you sure a scoped pointer works here? Because `QScopedPointer` needs the size of the type it holds and then we would need to include the header anyway.

Reply:
> Are you sure a scoped pointer works here?

@FlexW 

It actually works in other files. Search for `QScopedPointer` and check the header files. Half of them don't have the actual class header included in the header :)

Reply:
I see :) The key was to declare the destructor in `WelcomePage`.